tests/phpunit/tests/option/networkOption.php

```diff
@@ -699,4 +699,97 @@ public function data_stored_as_empty_string() {
 			'null'         => array( null ),
 		);
 	}
+
+	/**
+	 * Tests that update_network_option() adds a non-existent option that uses a filtered default value.
+	 *
+	 * @ticket 59360
+	 *
+	 * @covers ::update_network_option
+	 */
+	public function test_update_network_option_should_add_option_with_filtered_default_value() {
+		global $wpdb;
+
+		$option        = 'update_network_option_custom_default';
+		$default_value = 'default-value';
+
+		add_filter(
+			"default_site_option_{$option}",
+			static function () use ( $default_value ) {
+				return $default_value;
+			}
+		);
+
+		/*
+		 * For a non existing option with the unfiltered default of false, passing false here wouldn't work.
+		 * Because the default is different than false here though, passing false is expected to result in
+		 * a database update.
+		 */
+		$this->assertTrue( update_network_option( null, $option, false ), 'update_network_option() should have returned true.' );
+
+		$actual = $wpdb->get_row(
+			$wpdb->prepare(
+				"SELECT option_value FROM $wpdb->options WHERE option_name = %s LIMIT 1",
+				$option
+			)
+		);
+
+		$this->assertIsObject( $actual, 'The option was not added to the database.' );
+		$this->assertObjectHasProperty( 'option_value', $actual, 'The "option_value" property was not included.' );
+		$this->assertSame( '', $actual->option_value, 'The new value was not stored in the database.' );
```

Comment:
The database query will change depending on whether it's Single Site or Multisite:

```suggestion
		if ( is_multisite() ) {
				$table       = $wpdb->sitemeta;
				$name_field  = 'meta_key';
				$value_field = 'meta_value';
		} else {
				$table       = $wpdb->options;
				$name_field  = 'option_name';
				$value_field = 'option_value';
		}

		$actual = $wpdb->get_row(
			$wpdb->prepare(
				"SELECT %s FROM %s WHERE %s = 'foo' LIMIT 1",
				$value_field,
				$table,
				$name_field
			)
		);

		$this->assertIsObject( $actual, 'The option was not added to the database.' );
		$this->assertObjectHasProperty( $value_field, $actual, "The '$value_field' property was not included." );
		$this->assertSame( '', $actual->$value_field, 'The new value was not stored in the database.' );
```